### DISCUSSION TITLE
Modeling accounts / ledger entries?

### DISCUSSION THREAD

**Discussion by olalonde** [prisma/prisma#20674] [Q&A]
### Question

I am writing an accounting subsystem and have come up with the following data model:

```prisma
model Account {
  id                    Int                 @id @default(autoincrement())
  balance               Int                 @default(0)
  userId Int
  user User @relation(fields: [userId], references: [id])
  ledgerEntriesFrom  LedgerEntry[] @relation("LedgerEntryFromAccount")
  ledgerEntriesTo  LedgerEntry[] @relation("LedgerEntryToAccount")
  // How to have a "ledgerEntries" relation which returns all ledgerEntries, whether it's fromAccount or toAccount?
  // ledgerEntries LedgerEntry[]
}

model LedgerEntry {
  id                    Int                 @id @default(autoincrement())

  amount    Int // always positive

  fromAccountId Int
  toAccountId Int


  fromAccount             Account             @relation("LedgerEntryFromAccount", fields: [fromAccountId], references: [id])
  toAccount               Account             @relation("LedgerEntryToAccount", fields: [toAccountId], references: [id])
}
```

See the comment.

I was wondering if it was possible to have a `ledgerEntries LedgerEntry[]` on `Account` which would match all ledger entries, whether it is because of `fromAccountId` OR `toAccountId`.

Or if there is a better way to model this.

**Comment by nurul3101** [comment]
Hello @olalonde 👋 

At the moment, it's not possible to combine `ledgerEntriesFrom` and `ledgerEntriesTo` into a single `ledgerEntries`.
Each relationship needs to have a relation field on both sides of the model.

You could combine `ledgerEntries `it by doing something like this:
```ts
import { PrismaClient } from '@prisma/client';

const prisma = new PrismaClient();

async function getLedgerEntries(accountId: number) {
  const fromEntries = await prisma.ledgerEntry.findMany({
    where: { fromAccountId: accountId },
  });

  const toEntries = await prisma.ledgerEntry.findMany({
    where: { toAccountId: accountId },
  });

  return [...fromEntries, ...toEntries];
}

```

**Reply by olalonde** [reply]
Thanks for the info. I think this might work as well?

```ts
import { PrismaClient } from '@prisma/client';

const prisma = new PrismaClient();

async function getLedgerEntries(accountId: number) {
  const entries = await prisma.ledgerEntry.findMany({
    where: {
      OR: [
        { toAccountId: accountId },
        { fromAccountId: accountId }
      ]
    }
  });
  return entries;
}
```

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
